tests/handlers/records-read.spec.ts

```diff
@@ -604,6 +606,139 @@ export function testRecordsReadHandler(): void {
             expect(chatReadReply.status.code).to.equal(401);
             expect(chatReadReply.status.detail).to.contain(DwnErrorCode.ProtocolAuthorizationMissingRole);
           });
+
+          it('uses a contextRole to authorize a read', async () => {
+            // scenario: Alice creates a thread and adds Bob to the 'thread/participant' role. Alice writes a chat message.
+            //           Bob invokes the record to read in the chat message.
+
+            const alice = await DidKeyResolver.generate();
+            const bob = await DidKeyResolver.generate();
+
+            const protocolDefinition = threadRoleProtocolDefinition;
+
+            const protocolsConfig = await TestDataGenerator.generateProtocolsConfigure({
+              author: alice,
+              protocolDefinition
+            });
+            const protocolWriteReply = await dwn.processMessage(alice.did, protocolsConfig.message);
+            expect(protocolWriteReply.status.code).to.equal(202);
+
+            // Alice creates a thread
+            const threadRecord = await TestDataGenerator.generateRecordsWrite({
+              author       : alice,
+              recipient    : bob.did,
+              protocol     : protocolDefinition.protocol,
+              protocolPath : 'thread'
+            });
+            const threadRecordReply = await dwn.processMessage(alice.did, threadRecord.message, threadRecord.dataStream);
+            expect(threadRecordReply.status.code).to.equal(202);
+
+            // Alice adds Bob as a 'thread/participant' in that thread
+            const participantRecord = await TestDataGenerator.generateRecordsWrite({
+              author       : alice,
+              recipient    : bob.did,
+              protocol     : protocolDefinition.protocol,
+              protocolPath : 'thread/participant',
+              contextId    : threadRecord.message.contextId,
+              parentId     : threadRecord.message.recordId,
+            });
+            const participantRecordReply = await dwn.processMessage(alice.did, participantRecord.message, participantRecord.dataStream);
+            expect(participantRecordReply.status.code).to.equal(202);
+
+            // Alice writes a chat message in the thread
+            const chatRecord = await TestDataGenerator.generateRecordsWrite({
+              author       : alice,
+              protocol     : protocolDefinition.protocol,
+              protocolPath : 'thread/chat',
+              contextId    : threadRecord.message.contextId,
+              parentId     : threadRecord.message.recordId,
+            });
+            const chatRecordReply = await dwn.processMessage(alice.did, chatRecord.message, chatRecord.dataStream);
+            expect(chatRecordReply.status.code).to.equal(202);
+
+            // Bob invokes his 'participant' role to read the chat message
```

Comment:
Can we add the following checks:

1. Bob can also read the `participants` records, most importantly his own (but also others I guess) this is how I am imagining encryption keys can be hand off to and persisted for Bob.
2. Bob can also read the `thread` record, this is needed for bob to keep a copy of the thread in his own DWN.
2.

Reply:
Updated. Though for #1, Bob does not need to invoke his role to read his own role record. He is the recipient of the role record, and recipients are always authorized to read.